rails/before_filters.md

```diff
@@ -0,0 +1,91 @@
+Rails Before Filters
+==============================
+
+The purpose of before filters (now called before actions) is to filter access to
+controller actions. This makes them a good solution for things like
+authorization, but not a good one for things like populating instance variables.
+
+The Good
+--------------------
+
+DO the following
+
+```ruby
+class GoodController < ApplicationController
+  before_filter :authorize, :only => :show
+
+  def index
+  end
+
+  def show
+  end
+
+  private
+
+  def authorize
+    unless user_signed_in?
+      raise NotAuthorizedError
+    end
+  end
+end
+```
+
+No state changes, but in certain circumstances the entire action needs to be
+avoided.
+
+The Not So Good
+----------------------
+
+Avoid the following
+
+```ruby
+class NotSoGoodController < ApplicationController
+  before_filter :find_post, :only => [:show, :edit, :create]
```

Comment:
I'd ditch the :create here (and below where you use find_post in the create action) - it's actually a bit clearer how filters can be confusing when you only run a filter for some actions instead of all